crates/nu-command/src/filesystem/cd.rs

```diff
@@ -170,10 +172,16 @@ impl Command for Cd {
             stack.add_env_var("OLDPWD".into(), oldpwd)
         }
 
-        //FIXME: this only changes the current scope, but instead this environment variable
-        //should probably be a block that loads the information from the state in the overlay
+        // get a fake path ,is can get it , or not permission_denied , then can cd int
+        // the path
+        if permission_denied(PathBuf::from(format!("{}/enter", checkpath))) {
```

Comment:
So, you're using a fake file that may or may not exist. That's clever but I'm not sure it's the right thing to do.

Reply:
it may be ok if the fake file was some system-generated-unique-filename that would never exist. we need more eyes on this to approve or disapprove.

Reply:
@Decodetalkers can you change the word `enter` to some type of uniq system generated file name? I think, if we're going check for permission denied, we need to do it in such a way that there is really no possibility that the file exists and I think a unique name other than `enter` could do that.

Reply:
> @Decodetalkers can you change the word `enter` to some type of uniq system generated file name? I think, if we're going check for permission denied, we need to do it in such a way that there is really no possibility that the file exists and I think a unique name other than `enter` could do that.

Sorry, these days I am busy.. Now the problems is I don't know how to write the tests and what the uniq stystem generated file name should be...

Reply:
> @Decodetalkers can you change the word `enter` to some type of uniq system generated file name? I think, if we're going check for permission denied, we need to do it in such a way that there is really no possibility that the file exists and I think a unique name other than `enter` could do that.

Maybe "~"?, and will ~ available on macos and windows. However ~ is also available on linux, but always it should not be named as "~"

Reply:
There is probably an idiomatic rust way to unique temporary file names. If not, there are probably crates to do this type of thing too.